packages/react-instantsearch-core/src/core/createConnector.tsx

```diff
@@ -351,8 +356,15 @@ export function createConnectorWithoutContext(
 }
 
 const createConnectorWithContext =
-  (connectorDesc: ConnectorDescription) => (Composed: ReactType) => {
-    const Connector = createConnectorWithoutContext(connectorDesc)(Composed);
+  (connectorDesc: ConnectorDescription) =>
+  (
+    Composed: ReactType,
+    meta: WidgetMeta = { $$widgetType: 'custom-widget' }
```

Comment:
this is debatable, whether we either give a string default or an explicit null. I think a non-optional that allows string or undefined can be interesting too

Reply:
Can we get usage as easily when `$$widgetType` is undefined? If so, I think it's fine if we don't default it.

Reply:
I'd be in favor of requiring to fill it in, but if we do this, anyone using `createConnector` already and updating the library will have a TypeScript violation. That's arguably breaking.

Reply:
we can have it required internally, and still discuss on DefinitelyTyped what the type for externally is